lib/utils/application_insights.ts

```diff
@@ -0,0 +1,31 @@
+import * as ApplicationInsights from "applicationinsights";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { ServiceId } from "../api/definitions/ServiceId";
+
+export interface ITelemetryParams {
+  readonly operationId: NonEmptyString;
+  readonly operationParentId?: NonEmptyString;
+  readonly serviceId?: ServiceId;
+}
+
+export function getApplicationInsightsTelemetryClient(
+  params: ITelemetryParams,
+  commonProperties?: Record<string, string>
+): ApplicationInsights.TelemetryClient {
+  const telemetryClient = new ApplicationInsights.TelemetryClient();
+  // tslint:disable-next-line:no-object-mutation
+  telemetryClient.context.keys.operationId = params.operationId;
```

Comment:
mmh are you sure about this? from [the code](https://github.com/Microsoft/ApplicationInsights-node.js/blob/c992f39f17691f1842b225de30122856eb5acce3/Declarations/Contracts/Generated/ContextTagKeys.ts) it looks like this is the value of the `key` associated to the `operationId`, not the actual `operationId` value.

From [this example code](https://docs.microsoft.com/en-us/azure/application-insights/application-insights-correlation#role-name) it looks like the correct operation would be:

```ts
telemetryClient.context.tags[telemetryClient.context.keys.operationId] = params.operationId;
```

Reply:
also see [this code here](https://github.com/Microsoft/ApplicationInsights-node.js/blob/3abe097047d307441238e7e4a5afd5fa53fb4887/AutoCollection/HttpRequestParser.ts#L115)

Reply:
yes you're right

Reply:
awesome!